### PR TITLE
chore(assetcard,avatar,badge): remove mdx files

### PR DESCRIPTION
**Pull request by marissahuysentruyt**
## Description

This PR migrates any documentation from MDX files to the component's `*.stories.js` file. No loss of documentation should have occurred, but minor changes, like correcting story names to organize similar stories together or reorganizing content now that there is _not_ an MDX file or fixing sentence-casing, is acceptable. 

Components included are: 
- assetcard (there were some usage notes from the old docs site and some descriptions from the contributions site that were missing, so those were added)
- avatar
- badge (there was a missing sizing story, as well as a missing "notice" semantic variant, so those were added)

### Jira/Specs
[CSS-1047](https://jira.corp.adobe.com/browse/CSS-1047)

## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

### Validation steps

- [ ] Pull down the branch to run locally or visit the deploy preview.
- [ ] [Visit the asset card docs page](https://pr-3408--spectrum-css.netlify.app/?path=/docs/components-asset-card--docs).
- [ ] Verify the information on the docs page matches the [production docs page for asset card](https://opensource.adobe.com/spectrum-css/?path=/docs/components-asset-card--docs).
- [ ] Repeat the process for avatar (here's [the PR link](https://pr-3408--spectrum-css.netlify.app/?path=/docs/components-avatar--docs) to compare to [avatar's production](https://opensource.adobe.com/spectrum-css/?path=/docs/components-avatar--docs)) and badge (and this [PR link](https://pr-3408--spectrum-css.netlify.app/?path=/docs/components-badge--docs) to compare to [badge's production](https://opensource.adobe.com/spectrum-css/?path=/docs/components-badge--docs))
- [ ] Verify the badge sizing story renders on the docs page (this story is new).
- [ ] Verify the avatar testing preview correctly renders the "Disabled" test case without an anchor tag.

## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

### Regression testing

Validate:

1. The documentation pages for at least two other components are still loading, including:

- [ ] The pages render correctly, are accessible, and are responsive.

2. If components have been modified, VRTs have been run on this branch:

- [x] VRTs have been run and looked at.
- [x] Any VRT changes have been accepted (by reviewer and/or PR author), or there are no changes. (ran VRT against `main` and no changes were introduced!)

## To-do list

- [x] I have read the [contribution guidelines](/.github/CONTRIBUTING.md).
- [x] I have updated relevant storybook stories and templates.
- [x] If my change impacts **documentation**, I have updated the documentation accordingly.
- [ ] ✨ This pull request is ready to merge. ✨

### CODE REVIEW

**Review by cdransf — APPROVED** [review]
LGTM! ✨

File: components/badge/stories/badge.stories.js [hunk 1]
```diff
@@ -104,7 +107,12 @@ NonSemanticVariants.tags = ["!dev"];
 NonSemanticVariants.parameters = {
 	chromatic: { disableSnapshot: true },
 };
+NonSemanticVariants.storyName = "Non-semantic";
 
+/**
+ * Fixed positioning impacts the border radius of the badge component. The border radius is 0 along the fixed edge of the component. The actual component position is not represented on this page.
+ *
```
Comment (cdransf) [inline components/badge/stories/badge.stories.js:114]: _Tiny_ nit (feel free to ignore) — should we remove line 114 so there's no trailing line to match the other doc blocks? ✨

**Review by cdransf — APPROVED** [review]

File: components/badge/stories/badge.stories.js [hunk 2]
```diff
@@ -34,13 +33,14 @@ export default {
...
-			name: "Background color variants",
+			name: "Variants",
+			description: "Changes the badge's background color. The variant list includes both semantic and non-semantic options.",
 			type: { name: "string" },
 			table: {
 				type: { summary: "string" },
 				category: "Component",
 			},
-			options: ["neutral", "accent", "informative", "positive", "negative", "gray", "red", "orange", "yellow", "chartreuse", "celery", "green", "seafoam", "cyan", "blue", "indigo", "purple", "fuchsia", "magenta"],
+			options: ["neutral", "accent", "informative", "positive", "negative", "notice", "gray", "red", "orange", "yellow", "chartreuse", "celery", "green", "seafoam", "cyan", "blue", "indigo", "purple", "fuchsia", "magenta"],
```
Comment (castastrophe) [inline components/badge/stories/badge.stories.js:43]: Good catch, thanks for updating this.

File: components/badge/stories/template.js [hunk 3]
```diff
@@ -52,3 +53,13 @@ export const Template = ({
...
 };
+
+/* Displays icon-only, text-only, and icon-and-text badge options. */
+export const ContentOptions = (args, context) => Container({
+	withBorder: false,
+	content: html`
+		${Template(args, context)}
+		${Template({ ...args, iconName: undefined }, context)}
+		${Template({ ...args, label: undefined }, context)}
+	`
```
Comment (castastrophe) [inline components/badge/stories/template.js:64]: I mentioned this in the other PR, this is an alternate way to define the container content if you don't want to have to import the html literal here.

```suggestion
	content: [
		Template(args, context).
		Template({ ...args, iconName: undefined }, context),
		Template({ ...args, label: undefined }, context),
	]
```

**Review by castastrophe — APPROVED** [review]
Great work here, thanks for noticing the missing "notice" style in badge and getting that updated as well. I didn't check but if it's not also listed in the test case, can you make sure it's also added there too?

**Pull request merged**